upf/upf_proxy_accept.c

```diff
@@ -159,9 +159,13 @@ upf_tcp_lookup_connection (u32 fib_index, vlib_buffer_t * b, u8 thread_index,
       ip4 = vlib_buffer_get_current (b);
       tcp = ip4_next_header (ip4);
       tconn = session_lookup_connection_wt4 (fib_index,
+					     is_reverse ? &ip4->src_address :
 					     &ip4->dst_address,
+					     is_reverse ? &ip4->dst_address :
 					     &ip4->src_address,
+					     is_reverse ? tcp->src_port :
 					     tcp->dst_port,
+					     is_reverse ? tcp->dst_port :
 					     tcp->src_port,
```

Comment:
The excessive ?: looks ugly. Wouldn't it be better to have a pain `if ... else` only?

Reply:
Well, that adds another `if` nesting level which is also a bit dubious, esp. given that it's repeated for IPv4 and IPv6, and also that looked a bit cleaner before `indent` had split the lines ;) But ok, will redo with `if`s

Reply:
maybe use a wrapper around session_lookup_connection_wt4?

Reply:
In the end, another nested `if` didn't look that bad